lightning/src/ln/channelmonitor.rs

```diff
@@ -127,6 +127,21 @@ pub trait ManyChannelMonitor: Send + Sync {
 	fn fetch_pending_htlc_updated(&self) -> Vec<HTLCUpdate>;
 }
 
+/// SimpleArcChannelMonitor is useful for when you need a SimpleManyChannelMonitor with a static
+/// lifetime, e.g. when you're using lightning-net-tokio (since tokio::spawn requires parameters
+/// with static lifetimes). Other times you can afford a reference, which is more efficient, in
+/// which case SimpleRefChannelMonitor is the more appropriate type. Defining these type aliases
+/// prevents issues such as overly long function definitions.
+pub type SimpleArcChannelMonitor = Arc<SimpleManyChannelMonitor<OutPoint>>;
```

Comment:
I'm really torn on what to do here - SimpleManyChannelMonitor is great, but its also meant as a thing you need to put a wrapper around to store to disk. That probably needs to be clarified in the docs for ManyChannelMonitor, but maybe for now replace SimpleArcChannelMonitor with a template A: ManyChannelMonitor -> Arc<A>...I know, I contradicted myself here, sorry.

Reply:
Ah I'm not sure I understood this. What do you mean by 
>but its also meant as a thing you need to put a wrapper around to store to disk.

? Can't quite see the relevance. And, does "put a wrapper" mean put it into something that implements `Serializable`, or something similar?
 I replaced the SimpleArcChannelMonitor with a template within the SimpleArcChannelManager.

Reply:
As in users should almost certainly not be using SimpleManyChannelMonitor directly - they should be using MyCustomerManyChannelMonitor { in_memory_stuff: SimpleManyChannelMonitor } where the impl first writes the latest channel monitor to disk, then passes it to the in memory version. Thus, having a SimpleArcChannelMonitor is....not quite what we want (nor using SimpleManyChannelMonitor in any of the "suggested good default" types).